src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -182,7 +182,12 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
         await WaitForHttpSuccessOrThrow(dashboardUrl, TimeSpan.FromSeconds(DashboardWaitTimeInSeconds), cancellationToken).ConfigureAwait(false);
     }
 
+#if !DEBUG
     private const int DashboardWaitTimeInSeconds = 10;
+#else
+    // Allow extra time when building locally. Dashboard can take longer to start up because of dev binaries, debugger, etc.
+    private const int DashboardWaitTimeInSeconds = 30;
+#endif
```

Comment:
```suggestion
    // Allow extra time when building locally. Dashboard can take longer to start up because of dev binaries, debugger, etc
    private const int DashboardWaitTimeInSeconds = 30;
```